### DISCUSSION TITLE
Should I change the TimeZone from the MakerNote metadata?

### DISCUSSION THREAD

**Discussion by mjuanfe** [immich-app/immich#5659] [Q&A]
Hello,

I'm trying to combine my phone and DSLR photos, but with the DSLR ones appeared shifted in the timeline due to the different timezone, finally after some days I figured out to change the timezone with the exiftool and they placed correctly, but, although it did the trick I'm still wondering if I changed the appropriate metadata tag. 

**Here two commands that works as expected:**

 V2- `./exiftool "-timeZoneOffset=+9" -OffsetTime*=+09:00 -makernotes:timezone+=480 .\20230921_105944_nikonV2.JPG`

 V3- `./exiftool -makernotes:timezone+=480 .\20230921_105944_nikonV3.JPG`

**This one make the photo appear correctly in the timeline, however in the info view appears +01:00 where should be +09:00, isn't that strange? Or even could be a bug?**

V1- `./exiftool "-timeZoneOffset=+9" -OffsetTime*=+09:00 .\20230921_105944_nikonV1.JPG`

<img width="1868" alt="V1_" src="https://github.com/immich-app/immich/assets/5060381/4dc8833c-1e78-48df-8d1b-3338c664c1a8">

**So with this command:** 
 V3- `./exiftool -makernotes:timezone+=480 .\20230921_105944_nikonV3.JPG`

That is only changing the Timezone from the MakersNote metadata...
it's the correct way for Immich or could cause problems?

I'm about to change 50k photos and I don't want to mess it up

**Comment by jrasm91** [comment]
This specific picture has a GPS location so the timezones will probably come from that in this case.

**Reply by mjuanfe** [reply]
If the time zone is being based on location this picture should show GMT+9. 
That would be very nice, I wouldn't need to change any timezone related metadata.

All my DSLR photos have been manually geolocated with Lightroom,  but only with that none shows the corresponding timezone nor correctly placed in the timeline.

- The third command (V3) is enough to solve the problem, only wondering if is the right metadata tag for Immich(future proof, reliable...)
- And also curious about why the V1(shown picture) command makes the photo appear right placed in the timeline order but still showing GMT+1 in the picture info

I've been testing this with more pictures, restarting Immich, rescaning the library, reloading the metadata.... I always get the same results

**Reply by jrasm91** [reply]
I would be interested to see if anything changed if you ran "refresh metadata" on that specific pictures (available in the overflow menu on the web). If it still doesn't work, re-running it again with the log level set to verbose and then looking at the logs for more details would also help.

**Reply by mjuanfe** [reply]
I tried to refresh metadata again but nothing has changed.

Here relevant data of the verbose log:

V1 - [V1_Refresh_Metadata_log.txt](https://github.com/immich-app/immich/files/13664065/V1_Refresh_Metadata_log.txt)
```
"tz": "UTC+1",
"tzSource": "TimeZone",
"TimeZoneOffset": 9,
"OffsetTimeOriginal": "+09:00",
"OffsetTimeDigitized": "+09:00",
"TimeZone": "+01:00",
```

V3 - [V3_Refresh_Metadata_log.txt](https://github.com/immich-app/immich/files/13664066/V3_Refresh_Metadata_log.txt)
```
"tz": "UTC+9",
"tzSource": "TimeZone",
"TimeZone": "+09:00",
```


Do you know if is possible that this additional metadata tags in V1 is causing to place the photos correctly ordered in the timeline?

**Comment by everettcomstock** [comment]
I recently ran into a similar issue. I found that the photos I'd taken with my Canon DSLR didn't include the EXIF offset metadata fields like "OffsetTime". The photo files' created and modified date and time did not have any time  zone offset defined (no "-05:00" or "00:00"). However, the Canon Makernotes did include the Canon:TimeZone and Canon:TimeZoneCity fields. After running the following exiftool command that sets the Makernote values to GMT (and London... my camera model doesn't support UTC):
```
exiftool -n -Canon:TimeZone=0 -Canon:TimeZoneCity=20 IMAGE_NAME.JPG
```
Immich then correctly interpreted the photos as UTC.
